boltdesignsystem/bolt
# DS-93: convert to use typography css vars (Part 1)

## Jira

https://pegadigitalit.atlassian.net/browse/DS-93

## Summary

Converted a list of components to use typography CSS custom props instead of sass vars.

## Details

List of components updated:

- bolt-accordion
- bolt-blockquote
- bolt-button
- bolt-carousel
- bolt-code-snippet
- bolt-form
- bolt-headline
- bolt-li
- bolt-menu
- bolt-modal
- bolt-navbar
- bolt-nav-priority
- bolt-pagination
- bolt-popover
- bolt-progress-bar
- bolt-tabs
- bolt-toc
- bolt-tooltip
- bolt-trigger
- bolt-typeahead
- bolt-video

List of mixins/functions updated:

- bolt-font-family()
- bolt-font-size()
- bolt-font-weight()

## How to test

Run the branch locally and check the docs for all updated components. Make sure the typography is not different than what's on master, spacing might have 1~2px differences, overall it's very similar. Try to change the lang of the page to `ja` and see all fonts changed to system ui fonts.

## Release notes

### Deprecated SCSS mixins:

1. `bolt-font-family`
1. `bolt-font-size`
1. `bolt-font-weight`

### Deprecated SCSS functions

1. `bolt-line-height`

### Deprecated SCSS variables:

1. `$bolt-font-family-*`
1. `$bolt-font-size-*`
1. `$bolt-font-weight-*`
1. `$bolt-line-height-*`

### New typography vars usage

1. `var(--bolt-type-font-family-*)`: [headline, body, code]
1. `var(--bolt-type-font-weight-*)`: [light, regular, semibold, bold]
1. `var(--bolt-type-font-size-*)`: [xxsmall ~ xxxlarge]
1. `var(--bolt-type-line-height-*)`: [xxsmall ~ xxxlarge]

Note: font-size and line-height must be used in tandem. For example, small font-size needs to be paired with small line-height.

#### Math

```scss
font-size: calc(var(--bolt-type-font-size-medium) + 0.05vw);
```

Note: all math with CSS custom properties must be done with `calc()`.
